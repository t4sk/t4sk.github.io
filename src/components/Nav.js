import React from "react"
import PropTypes from "prop-types"
import { withRouter } from "react-router-dom"
import { Menu, Image } from "semantic-ui-react"
import { BACKGROUND_COLOR } from "../constants"
import logo from "./logo.svg"

export function Nav(props) {
  const { location } = props

  return (
    <Menu
      borderless
      fluid
      inverted
      fixed="top"
      style={{
        // boxShadow: "none",
        // borderBottom: "none",
        backgroundColor: BACKGROUND_COLOR,
      }}
      size="large"
    >
      <Menu.Item header as="a" href="/#top" style={{ padding: 0, width: 300 }}>
        <Image
          src={logo}
          style={{
            height: 48,
            backgroundColor: BACKGROUND_COLOR,
            margin: 5,
            borderRadius: 4,
          }}
          alt="logo"
        />
        Smart Contract Programmer
      </Menu.Item>

      <Menu.Menu position="right">
        <Menu.Item
          as="a"
          href="/#courses"
          active={location.hash === "#courses"}
        >
          Courses
        </Menu.Item>
        <Menu.Item as="a" href="/#about" active={location.hash === "#about"}>
          About
        </Menu.Item>
        <Menu.Item
          as="a"
          href="/#contact"
          active={location.hash === "#contact"}
        >
          Contact
        </Menu.Item>
      </Menu.Menu>
    </Menu>
  )
}

Nav.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
    hash: PropTypes.string.isRequired,
  }).isRequired,
}

export default withRouter(Nav)
