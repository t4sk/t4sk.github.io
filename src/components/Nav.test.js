import React from "react"
import { shallow } from "enzyme"
import { Nav } from "./Nav"

const props = {
  location: {
    pathname: "/",
    hash: "",
  },
}

test("it renders", () => {
  const component = shallow(<Nav {...props} />)

  expect(component).toMatchSnapshot()
})

test("tutorials active", () => {
  const component = shallow(
    <Nav {...props} location={{ ...props.location, hash: "#tutorials" }} />
  )

  expect(component).toMatchSnapshot()
})

test("about active", () => {
  const component = shallow(
    <Nav {...props} location={{ ...props.location, hash: "#about" }} />
  )

  expect(component).toMatchSnapshot()
})

test("contact active", () => {
  const component = shallow(
    <Nav {...props} location={{ ...props.location, hash: "#contact" }} />
  )

  expect(component).toMatchSnapshot()
})
