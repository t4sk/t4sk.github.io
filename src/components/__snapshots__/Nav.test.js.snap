// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<Menu
  borderless={true}
  fixed="top"
  fluid={true}
  inverted={true}
  size="large"
  style={Object {}}
>
  <Container>
    <MenuItem
      as="a"
      header={true}
      href="/"
      style={
        Object {
          "padding": 0,
          "paddingRight": 15,
        }
      }
    >
      <Image
        alt="logo"
        as="img"
        src="logo.svg"
        style={
          Object {
            "backgroundColor": "#1b1c1d",
            "borderRadius": 4,
            "height": 48,
            "margin": 5,
          }
        }
        ui={true}
      />
      Smart Contract Programmer
    </MenuItem>
    <MenuMenu
      position="right"
    />
  </Container>
</Menu>
`;
